editor/filesystem_dock.cpp

```diff
@@ -52,11 +52,21 @@
 
 FileSystemDock *FileSystemDock::singleton = nullptr;
 
-Ref<Texture2D> FileSystemDock::_get_tree_item_icon(bool p_is_valid, String p_file_type) {
+Ref<Texture2D> FileSystemDock::_get_tree_item_icon(bool p_is_valid, String p_file_type, EditorFileSystemDirectory *p_dir, int p_file_index) {
 	Ref<Texture2D> file_icon;
 	if (!p_is_valid) {
 		file_icon = get_theme_icon(SNAME("ImportFail"), SNAME("EditorIcons"));
 	} else {
+		if (p_file_type == "Resource") {
```

Comment:
This assumes that all user-defined resources will be things that extend `Resource`. If I chose to implement a script extending, say, `Texture`, then I would be unable to make use of this. Nor if I chose to define a second type extending my first one. So,  rather than checking for an exact match, you have to perform another inheritance check, akin to the process I described in my earlier comment w/ ScriptServer, EditorData, and ClassDB.

Reply:
@Atlinx Oh, also, when referring to the String values of type names, it's best to use `<type>::get_class_static()` rather than a hardcoded string. So, in this instance, you'd use `Resource::get_class_static()`.